scripts/sysman/ras.yml

```diff
@@ -173,6 +193,27 @@ returns:
         - "Don't have permissions to set thresholds."
 --- #--------------------------------------------------------------------------
 type: function
+desc: "Get the number of RAS error category associated with the error set handle"
+version: "1.7"
+class: $sRas
+name: EnumRasErrorCategory
+details:
+    - "A RAS error set has a varying number of RAS error category associated depends on the platform type(see $s_ras_state_t for a breakdown)"
+    - "If the function completes successfully but pCount is set to 0, RAS error category is not available on handle."
+    - "The application may call this function from simultaneous threads."
+    - "The implementation of this function should be lock-free."
+params:
+    - type: $s_ras_handle_t
+      name: hRas
+      desc: "[in] Handle for the component."
+    - type: "uint32_t*"
```

Comment:
usually pCount interfaces have in the same method the array. Could you elaborate on how this should be used? could we add some notes in programming guide?

Reply:
This function is used returns the #of error category supported, and the used to dynamically allocate the array in $s_ras_ext_state_t.pCategory, when calling zesRasGetState. We could also implment like a python dictionary style {key, value}, where each error categery to a error count {zes_ras_error_cat_t, error_count}. May be a cleaner solution. In that case, we need to extend the zesRasGetState function. Let me know. Yes, we need this to be added to the programer guide.

Reply:
Perhaps rename `pCount` to `pNumCategories`